lib/resources/http.rb

```diff
@@ -165,7 +165,7 @@ def status
 
         def body
           run_curl
-          @body
+          @body.strip
```

Comment:
What happens if body is nil?

Reply:
This would be the perfect spot for Ruby 2.3+'s safe navigation operator:
https://www.ruby-lang.org/en/news/2015/12/25/ruby-2-3-0-released/

```irb
➜  irb
Welcome to IRB. You are using ruby 2.4.2p198 (2017-09-14 revision 59899) [x86_64-darwin16]. Have fun ;)
>> nil&.strip #=> nil
>> 'foo'&.strip #=> "foo"
>> 'bar
'>
'>
'> '&.strip #=> "bar"
```

Looks like [InSpec is currently locked on Ruby 2.1+](https://github.com/chef/inspec/blob/master/inspec.gemspec#L27)....what is InSpec's policy surrounding minimum Ruby version?

Reply:
I don't know that we have a formal one, but probably should. Informally it's been "support the oldest version on normal maintenance" which is now 2.3 anyways. I'd be in favor of bumping to 2.3 in the gemspec. We already plan on bumping to 2.4 in the omnibus builds and that's on my backlog.

@chris-rock @arlimus thoughts?

Reply:
I agree that bumping to 2.4 for omnibus and testing is a good idea. I do not agree that the inspec ruby code should assume that ruby 2.4 is there. In case you need it, the resource need to be able fail if ruby version x is not available.

Reply:
So what is the minimum Ruby version we can assume? Is it OK to assume 2.3 which is _the oldest version on normal maintenance_ as @adamleff mentioned?